src/serac/physics/tests/solid_multi_material.cpp

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2019-2024, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/solid_mechanics.hpp"
+
+#include "axom/slic/core/SimpleLogger.hpp"
+#include <gtest/gtest.h>
+
+#include "serac/mesh/mesh_utils.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/physics/materials/solid_material.hpp"
+#include "serac/infrastructure/initialize.hpp"
+#include "serac/infrastructure/terminator.hpp"
+
+namespace serac {
+
+template <int dim>
+tensor<double, dim> average(std::vector<tensor<double, dim>>& positions)
+{
+  tensor<double, dim> total{};
+  for (auto x : positions) {
+    total += x;
+  }
+  return total / double(positions.size());
+}
+
+TEST(Solid, MultiMaterial)
+{
+  /*
+   * Checks multi material case with the following uniaxial problem:
+   *              MATERIAL 1            MATERIAL 2
+   *               E = 1                 E = 2
+   * u = 0   --------------------|-------------------- stress = 1
+   *
+   * Solution:
+   * strain =       1                    0.5
+   *
+   */
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  constexpr int p                   = 2;
+  constexpr int dim                 = 3;
+  int           serial_refinement   = 0;
+  int           parallel_refinement = 0;
+
+  // Create DataStore
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, "solid_mechanics_multimaterial");
+
+  constexpr double L      = 8.0;
+  constexpr double W      = 1.0;
+  constexpr double H      = 1.0;
+  constexpr double VOLUME = L * W * H;
+
+  auto mesh = mesh::refineAndDistribute(buildCuboidMesh(8, 1, 1, L, W, H), serial_refinement, parallel_refinement);
+
+  const std::string mesh_tag{"mesh"};
+
+  auto& pmesh = serac::StateManager::setMesh(std::move(mesh), mesh_tag);
+
+  // _solver_params_start
```

Comment:
These tags are used in the sphinx documentation and are not needed here. Looking at the code these tags have bled into other files and should only be in `serac/physics/tests/solid.cpp`.

Reply:
Good catch, I didn't even see those.